packages/voting-contract/src/lib.rs

```diff
@@ -136,7 +137,13 @@ where
             vote,
         }),
     })?;
-
+    BALLOTS_BY_VOTER.update(deps.storage, (&info.sender, proposal_id), |bal| match bal {
+        Some(_) => Err(ContractError::AlreadyVoted {}),
+        None => Ok(Ballot {
+            weight: vote_power,
+            vote,
+        }),
+    })?;
```

Comment:
@ethanfrey 
> I think this is fine for this PR as they are only written once and never updated.

...although they are updated.

Does that change your position here?